### PR TITLE
Integration test

### PR DESCRIPTION
**Pull request by marcus1337**
Added code to output a plot image showing how algorithms compare against the original Percolator version in terms of fractions, where a fraction value of 1 means no difference, 2 means 2 times slower and 0.5 means twice as fast.

### CODE REVIEW

File: data/integration_tests/percolator/IntegrationTest_Percolator_SpeedGraph.py [hunk 1]
```diff
@@ -141,31 +142,106 @@ def getXValues(file):
...
-    chart.savefig("{}/{}.png".format(pathToOutputData, outputFilename + "Line"))
-    chart.close()
+    savePlot(outputFilename + "Line")
+
+def getMaxAndMinPercentage(data):
+    minVal = 999999999999
+    maxVal = -99999999999
+    for value in data.values():
+        percentages = list(value.values())
+        maxVal = max(maxVal, math.ceil(max(percentages))) #max av all kommande data
```
Comment (BR903) [inline data/integration_tests/percolator/IntegrationTest_Percolator_SpeedGraph.py:165]: (Translate to English?)

File: data/integration_tests/percolator/IntegrationTest_Percolator_SpeedGraph.py [hunk 2]
```diff
@@ -141,31 +142,106 @@ def getXValues(file):
...
+    for key in data.keys():
+        relativeValues = data[key].values()
+        percentages = [y/x for x, y in zip(referenceValues, relativeValues)] #Assume it's impossible for any run-time to be 0.
+        percentages = [ round(elem, 3) for elem in percentages ]
+        data[key].update(zip(data[key], percentages))
+
+def addRelativeLines(data, xTicks, colors):
+    counter = 1
+    for key in data.keys():
+        #print(data[key].values())
```
Comment (BR903) [inline data/integration_tests/percolator/IntegrationTest_Percolator_SpeedGraph.py:179]: Delete commented-out code before committing.

File: data/integration_tests/percolator/IntegrationTest_Percolator_SpeedGraph.py [hunk 3]
```diff
@@ -141,31 +142,106 @@ def getXValues(file):
...
+    firstKey = next(dataIter)
+    referenceValues = tmpData[firstKey].values()
+    xTicks = list(tmpData[firstKey].keys())
+    del tmpData[firstKey]
+    return referenceValues, xTicks
+
+def nextClosePowerOf2(x):  
+    if x >= 1:
+        x = int(x)
+        return 1 if x == 0 else 2**(x - 1).bit_length()
```
Comment (BR903) [inline data/integration_tests/percolator/IntegrationTest_Percolator_SpeedGraph.py:194]: Much cheaper (and less roundabout) to use left-shift in place of exponentiation: `1 << (x-1).bit_length()`.

File: data/integration_tests/percolator/IntegrationTest_Percolator_SpeedGraph.py [hunk 4]
```diff
@@ -141,31 +142,106 @@ def getXValues(file):
...
+
+    minDivisionOf2 = 1
+    while minDivisionOf2 >= x:
+        minDivisionOf2 /= 2
+    x = minDivisionOf2
```
Comment (BR903) [inline data/integration_tests/percolator/IntegrationTest_Percolator_SpeedGraph.py:199]: Just return minDivisorOf2 instead of modifying x.

**Review by BR903 — APPROVED** [review]

**Comment by marcus1337** [comment]
Made some further modifications by adding more gridlines. I also now clearly point out the the y-axis is logarithmic (base 2).

**Comment by marcus1337** [comment]
Discovered some numerical errors for input values less than 1 because the y axis scales differently. Will fix this before sending another pull request.

**Pull request merged**